build_runner/lib/src/server/server.dart

```diff
@@ -64,15 +66,16 @@ class ServeHandler implements BuildState {
     _state.currentBuild.then((_) => _warnForEmptyDirectory(rootDir));
     var cascade = new shelf.Cascade()
         .add(_blockOnCurrentBuild)
-        .add((shelf.Request request) {
+        .add((shelf.Request request) async {
       if (request.url.path == _performancePath) {
         return _performanceHandler(request);
       }
       if (request.url.path.startsWith(_graphPath)) {
-        return _assetGraphHandler.handle(
-            request.change(path: _graphPath), rootDir);
+        var graphHandler = await _assetGraphHandler;
```

Comment:
Is there an sdk bug we can point at for this?

Reply:
I wouldn't call this a bug but it definitely is awkward. It seems related to https://github.com/dart-lang/sdk/issues/28768